### PR TITLE
fix #18231

### LINKED ISSUE #18231: Persist git branches that have been recently checked out
I use "Git: Checkout" a lot and it would be really useful to have a section of recent branches at the top, like how cmd+p always has "recently opened" above all file results.

### PR DESCRIPTION
**Pull request by ohana54**
To simplify things, I chose to keep the usage list in memory.
It can be persisted to disk (I have stashed code for that) but it requires more work, specifically - where do we keep the file. I thought to put it in the `.git` folder but I don't know if it's best practice.

### CODE REVIEW

File: extensions/git/src/commands.ts [hunk 1]
```diff
@@ -139,6 +139,7 @@ const ImageMimetypes = [
 export class CommandCenter {
 
 	private disposables: Disposable[];
+	private checkoutItemsUsageList: Array<string>;
```
Comment (joaomoreno) [inline extensions/git/src/commands.ts]: Let's store this in [`workspaceState`](https://github.com/Microsoft/vscode/blob/c7880659857c8d62a70fe001d8539ab987849767/src/vs/vscode.d.ts#L4085) instead.

**Review by joaomoreno — CHANGES_REQUESTED** [review]
There are quite a few `any` types and missing function return  types.

Also, I suggest to simply record timestamps of checkout, and then sort branches given their last checkout times.

File: extensions/git/src/commands.ts (hunk 1 above)
Comment (ohana54) [reply extensions/git/src/commands.ts]: Sure, how do I do that? Didn't see any mentions of `workspaceState` in this file. Could you please point me to a similar usage or a written manual?

**Comment by ohana54** [comment]
@joaomoreno Thanks for the review, I changed the implementation as you suggested and on the way fixed the types. It's my first contribution to a TS project, so let me know if there are more type changes need to be made.

**Comment by joaomoreno** [comment]
Hi @ohana54, actually since https://github.com/Microsoft/vscode/pull/38563, the sort order is much better than lexicographical. While it's not exactly what you were aiming for, it still achieves a very good equivalent and it is a much cleaner solution as it doesn't require any state storage on our side.

**Pull request closed without merging**